libflux/core/src/semantic/nodes.rs

```diff
@@ -608,11 +608,12 @@ impl StringExpr {
             if let StringExprPart::Interpolated(ref mut ip) = p {
                 let (e, cons) = ip.expression.infer(env, f)?;
                 constraints.append(&mut Vec::from(cons));
-                constraints.push(Constraint::Equal {
-                    exp: MonoType::String,
+                constraints.push(Constraint::Kind {
+                    exp: Kind::Stringable,
                     act: ip.expression.type_of(),
                     loc: ip.expression.loc().clone(),
                 });
+                //TODO: add Kind::Stringable constraint
```

Comment:
Is there an open issue for this?

Reply:
Good catch, it was a mental note I was making and should be deleted.